velox/exec/Aggregate.h

```diff
@@ -313,17 +314,31 @@ class Aggregate {
   std::vector<vector_size_t> pushdownCustomIndices_;
 };
 
+// 'argTypes' can be either raw input types or intermediate type.  'resultType'
+// is either final output or intermdiate type.  'signature' is the function
+// signature from planner, only non-null if the information is provided.  Even
+// 'signature' is non-null, it is not guaranteed to be able to cast to
+// 'AggregateFunctionSignature'.
 using AggregateFunctionFactory = std::function<std::unique_ptr<Aggregate>(
     core::AggregationNode::Step step,
     const std::vector<TypePtr>& argTypes,
-    const TypePtr& resultType)>;
+    const TypePtr& resultType,
+    const FunctionSignaturePtr& signature)>;
```

Comment:
It would be nice to explain the problem and proposed solution in the PR description. 

I wonder if we can avoid changing registry API. An aggregate function created using raw input types should accept both raw input and intermediate results. In case when we know the raw input types, we can create Aggregate function instance using these, then add intermediate results.

Reply:
The description is updated.

The problem is some functions are only created based on intermediate type (from worker's perspective).  In this case raw input types must be passed from planner because the worker does not have this information.